src/support/modify.c

```diff
@@ -434,8 +435,9 @@ __wt_modify_reconstruct_from_upd_list(WT_SESSION_IMPL *session, WT_CURSOR_BTREE
         WT_RET_SUB(session, WT_ROLLBACK, WT_MODIFY_READ_UNCOMMITTED,
           "Read-uncommitted readers do not support reconstructing a record with modifies.");
 retry:
-    /* Construct full update */
-    __wt_update_vector_init(session, &modifies);
+    /* Reconstruct full update */
+    __wt_update_vector_free(&modifies);
```

Comment:
It would be worth adding a comment here describing why we need to call the free function here. It's not directly obvious if you miss the `retry` goto statement